archive/multisource.go

```diff
@@ -0,0 +1,181 @@
+package archive
+
+import (
+	"context"
+	"io"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/driver"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/scanner"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zio"
+	"github.com/brimsec/zq/zio/detector"
+	"github.com/brimsec/zq/zio/zngio"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+// A SpanInfo is a logical view of the records within a time span, stored
```

Comment:
I don't think 'SpanInfo' is a good name for this type anymore, but before I change it & the relevant comments, wanted to get buy in: I was thinking of calling this a 'segment'. (It doesn't need to be exported anymore.)

Reply:
ok, I unexported, so it's just `spanInfo` for now.